src/SAML2/Binding/AbstractHttpBinding.php

```diff
@@ -0,0 +1,298 @@
+            ));
+        }
+
+        return $message;
+    }
+
+    /**
+     * @param Request $request
+     * @return \SAML2_LogoutResponse
+     * @throws \AdactiveSas\Saml2BridgeBundle\Exception\InvalidArgumentException
+     */
+    public function receiveSignedLogoutResponse(Request $request){
+        $message = $this->receiveSignedMessage($request);
+
+        if (!$message instanceof \SAML2_LogoutResponse) {
+            throw new InvalidArgumentException(sprintf(
+                'The received request is not an LogoutRequest, "%s" received instead',
+                substr(get_class($message), strrpos($message, '_') + 1)
+            ));
+        }
+
+        return $message;
+    }
+
+    /**
+     * @param Request $request
+     * @return \SAML2_AuthnRequest
+     * @throws \AdactiveSas\Saml2BridgeBundle\Exception\InvalidArgumentException
+     */
+    public function receiveUnsignedAuthnRequest(Request $request){
+        $message = $this->receiveUnsignedMessage($request);
+
+        if (!$message instanceof \SAML2_AuthnRequest) {
+            throw new InvalidArgumentException(sprintf(
+                'The received request is not an AuthnRequest, "%s" received instead',
+                substr(get_class($message), strrpos($message, '_') + 1)
+            ));
+        }
+
+        return $message;
+    }
+
+    /**
+     * @param Request $request
+     * @return \SAML2_LogoutRequest
+     * @throws \AdactiveSas\Saml2BridgeBundle\Exception\InvalidArgumentException
+     */
+    public function receiveUnsignedLogoutRequest(Request $request){
+        $message = $this->receiveUnsignedMessage($request);
+
+        if (!$message instanceof \SAML2_LogoutRequest) {
+            throw new InvalidArgumentException(sprintf(
+                'The received request is not an LogoutRequest, "%s" received instead',
+                substr(get_class($message), strrpos(get_class($message), '_') + 1)
+            ));
+        }
+
+        return $message;
+    }
+
+    /**
+     * @param Request $request
+     * @return \SAML2_LogoutResponse
+     * @throws \AdactiveSas\Saml2BridgeBundle\Exception\InvalidArgumentException
+     */
+    public function receiveUnsignedLogoutResponse(Request $request){
+        $message = $this->receiveUnsignedMessage($request);
+
+        if (!$message instanceof \SAML2_LogoutResponse) {
+            throw new InvalidArgumentException(sprintf(
+                'The received request is not an LogoutRequest, "%s" received instead',
+                substr(get_class($message), strrpos(get_class($message), '_') + 1)
+            ));
+        }
+
+        return $message;
+    }
+
+    /**
+     * @param Request $request
+     * @return \SAML2_Message
+     * @throws \AdactiveSas\Saml2BridgeBundle\Exception\BadRequestHttpException
+     */
+    public function receiveSignedMessage(Request $request)
+    {
+        $query = $this->getReceivedMessageQueryString($request);
```

Comment:
Maybe we should rename `getReceivedMessageQueryString` to something more abstract because of the `QueryString` wording.